1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,346 @@
+- Note: Windows PowerShell 5.1 and PowerShell 7.0 will only support prediction source **History**.
+  PowerShell 7.1+ will support prediction sources **History**, **Plugin**, and **HistoryAndPlugin**
+
+## Goals/Non-Goals
+
+ Goals:
+
+- Provide suggestions from PSReadLine history
+- The user will be able to enable and disable Predictive IntelliSense
+- The user will be able to change between InlineView and ListView
+- The user will have keyboard shortcuts to navigate and edit a prediction
+- The user will be able to register additional providers when desired
+- The user may customize the color of a prediction to support accessibility needs
+
+Non-goals:
+
+- Due to the relatively short length of a session compared to your PSReadline history,
+  PowerShell session history is not part of the historical suggestion
+- The extension framework doesn't allow a plugin implementation to call into the Runspace where
+  PSReadLine is running when providing predictive suggestions.
+- VSCode implementation. We agree this is important for a consistent experience and are working to
+  bring this feature in a future release, but the user experience for VS Code is outside the scope of this RFC.
+
+## Specification
+
+The proposal is to add Predictive IntelliSense and an extension model to improve the user
+interactive experience.
+
+### Key Design Considerations Predictive IntelliSense
+
+Predictive IntelliSense displays the most recent match command completion from command history. The
+prediction may change if a better match is found as the user types additional information. The
+predicted command may be accepted, edited, or ignored.
+
+![Prediction - Historical](./media/pred-wlc.png)
+
+### Install and Remove Predictive IntelliSense
+
+Predictive IntelliSense is implemented in both the PowerShell engine and presented through the
+PSReadLine module. To receive the benefits of Predictive IntelliSense based on your PowerShell
+version, download and install the PSReadLine module from PSGallery
+
+The current release is PSReadLine 2.1.0 Beta 2:
+
+```powershell
+Install-Module PSReadLine -RequiredVersion 2.1.0-beta2 -AllowPrerelease
+```
+
+It is possible to unload or uninstall the PSReadLine module to disable Predictive IntelliSense. This
+creates a negative impact to customers as it would remove all features of PSReadLine. For this
+reason, Predictive IntelliSense has its own PSReadLine configuration to enable and disable
+predictions.
+
+### Enabling and Disabling Predictions
+
+By default, the feature Predictive IntelliSense is disabled. The impact of enabling this feature by
+default may cause new and existing users to become confused at the shell prompt. The feature is
+enabled and disabled with a PSReadLine command Set-PSReadLineOption that is executed at the shell
+prompt or in the user's Profile.
+
+Predictive IntelliSense in PSReadLine 2.1.0-beta2 currently supports the following arguments for
```

Comment:
I know that beta2 specifically had impact to users throwing an exception that a type wasn't being found.  It would be good to explicitly call out the user experience if they try to enable predictions on a PS version that doesn't support it

Reply:
Good call. I added a note at the bottom of the Install and Remove section. [!NOTE] 
When upgrading PowerShell preview versions, it's important to upgrade to the corresponding
version of PSReadLine. Incorrect versions will result in an error when PowerShell starts.

Reply:
"Some beta versions may result in an error when PowerShell starts."